### DISCUSSION TITLE
Have Khoj understand metadata about indexed files and provide high-level insights

### DISCUSSION THREAD

**Discussion by sabaimran** [khoj-ai/khoj#396] [Ideas]
A common use case might be to summarize metadata across files, as if I, as the AI agent, can see your file system, summarize across notes, and provide aggregated insights. Our architecture right now is currently very file-specific, but it would be pretty neat if we could start providing holistic results.

[Backlinking relevant discussion in Discord](https://discord.com/channels/1112065956647284756/1112069753549373480/1135489974301642822).

**Comment by zakirullin** [comment]
Great! I'll copy the exact use cases from that thread.

I have md files under journal directory named as follows:
`July 2022.md`
`May 2023.md`

So on. This is journaling. And I want to execute a query like that:
"Could you please extract common behaviour patterns from my records under journal folder?"

Also queries like: "Could you please extract main topics from all the notes under architecture folder?" would be insightful as well.
